util/shmctl.c

```diff
@@ -1226,6 +1216,18 @@ int main(int argc, char **argv)
 	printf("commands:\n");
 	for( i = 0 ; cmd[i].func != NULL ; ++i )
 	    printf("\t%-15s%s\n", cmd[i].cmd, cmd[i].descript);
+	return 0;
     }
-    return 0;
+    if( cmd[i].func == SHMinit ){
+	/* in this case, do NOT run attach_SHM here.
+	   because uhash_loader is not run yet.      */
+	return SHMinit(argc - 1, &argv[1]);
+    }
+
+    attach_SHM();
+    /* shmctl doesn't need resolve_boards() first */
+    //resolve_boards();
+    resolve_garbage();
+    resolve_fcache();
+    return cmd[i].func(argc - 1, &argv[1]);
```

Comment:
The flow becomes:
1. if argc >= 2: find the corresponding cmd.
2. if argc == 1: print all commands and return.
3. special case for SHMinit.
4. other cmds.

Suggestion:
1. if argc == 1: print and return.
2. find the corresponding cmd.
3. special case for SHMinit.
4. other cmds.

Reply:
Actually we do need to find the corresponding cmd first
because it's possible that the cmd (argv[1]) is an invalid cmd.

The original flow is good.